### ISSUE TITLE
CloseDatabase prior to CMFE causes parallel engine to crash during CMFE.

### ISSUE THREAD

**Issue by aowen87** [visit-dav/issues-test#2145]
Kathleen reported that the export_db.py test was failing on Windows because it tried to remove files that were open in the engine. I added the appropriate CloseDatabase() calls to the test to try and make it more Windows-friendly. Things seemed okay until I ran the test case in parallel. In parallel, the presence of certain CloseDatabase() calls causes the engine to crash while processing the load balancing for the CMFE. This happens on Mac and Linux. I commented out all the CloseDatabase() lines in export_db.py for the time-being. This bug should be considered done when the engine does not crash in parallel and the export_db.py test passes in parallel with the calls to CloseDatabase().

<pre>
import os, sys

def silo_data_path(filename):
    return os.path.join("/usr/gapps/visit/data", filename)

def cmfe_silo_data_path(filename):
    return silo_data_path(filename)

OpenDatabase(silo_data_path("globe.silo"))
AddPlot("Pseudocolor", "t")
DrawPlots()
DeleteAllPlots()
CloseDatabase(silo_data_path("globe.silo"))   ### If we do not have this then we're fine. Closing the database hoses CMFE.

OpenDatabase(silo_data_path("wave.visit"))
DefineScalarExpression("cmfe", "conn_cmfe(coord(<%s:quadmesh>)[1], quadmesh)" % cmfe_silo_data_path("wave0020.silo"))
AddPlot("Pseudocolor", "cmfe")
DrawPlots()
SaveWindow()

sys.exit(0)
</pre>

Here's the command I used on surface: ./visit -cli -np 2 -l srun -nowin -s crash.py

Here's the stack I get when I run the parallel engine under GDB:
<pre>
Program received signal SIGSEGV, Segmentation fault.
0x0000000000000019 in ?? ()
Missing separate debuginfos, use: debuginfo-install freetype-2.3.11-15.el6_6.1.x86_64 keyutils-libs-1.4-5.el6.x86_64 krb5-libs-1.10.3-37.1chaos.ch5.3.x86_64 libICE-1.0.6-1.el6.x86_64 libSM-1.2.1-2.el6.x86_64 libX11-1.6.0-2.2.el6.x86_64 libXau-1.0.8-2.1.ch5.2.x86_64 libXext-1.3.2-2.1.el6.x86_64 libXt-1.1.4-6.1.el6.x86_64 libcom_err-1.41.12-21.el6.x86_64 libgcc-4.4.7-11.el6.x86_64 libgfortran-4.4.7-11.el6.x86_64 libgomp-4.4.7-11.el6.x86_64 libibumad-1.3.9-1.el6.x86_64 libibverbs-1.1.8-3.el6.x86_64 libicu-4.2.1-11.el6.x86_64 libipathverbs-1.3-3.el6_5.x86_64 libmlx4-1.0.6-6.el6.x86_64 libmthca-1.0.6-3.el6.x86_64 libnl-1.1.4-2.el6.x86_64 librdmacm-1.0.18.1-1.el6.x86_64 libselinux-2.0.94-5.8.el6.x86_64 libstdc++-4.4.7-11.el6.x86_64 libuuid-2.17.2-12.18.el6.x86_64 libxcb-1.9.1-2.el6.x86_64 libxml2-2.7.6-17.el6_6.1.x86_64 mesa-libGLU-10.1.2-2.el6.x86_64 munge-libs-0.5.11-1.ch5.1.1.x86_64 numactl-2.0.9-2.el6.x86_64 openssl-1.0.1e-30.el6_6.11.x86_64 zlib-1.2.3-29.el6.x86_64
(gdb) where
#0  0x0000000000000019 in ?? ()
#1  0x00002aaaaf596fa2 in avtDatabase::GetMetaData(int, bool, bool, bool) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtdatabase_par.so
#2  0x00002aaaab5db6d4 in LoadBalancer::GetMeshName(ref_ptr<avtContract>, int) () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#3  0x00002aaaab5dc0ae in LoadBalancer::Reduce(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#4  0x00002aaaab5deb17 in ReduceCallback(void*, ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#5  0x00002aaab07c34b8 in avtOriginatingSource::BalanceLoad(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#6  0x00002aaab07c4589 in avtOriginatingSource::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#7  0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#8  0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#9  0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#10 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#11 0x00002aaaada602a6 in avtCMFEExpression::Execute() () from /g/g17/whitlocb/Development/trunk/src/lib/libavtexpressions_par.so
#12 0x00002aaab07a4082 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#13 0x00002aaaadb098af in avtExpressionEvaluatorFilter::Execute() () from /g/g17/whitlocb/Development/trunk/src/lib/libavtexpressions_par.so
#14 0x00002aaab07a4082 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#15 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#16 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#17 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#18 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#19 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#20 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#21 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#22 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#23 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#24 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#25 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#26 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#27 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#28 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#29 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#30 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#31 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#32 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#33 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#34 0x00002aaab07b2c0d in avtDataObjectSink::UpdateInput(ref_ptr<avtContract>) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#35 0x00002aaab07a3f23 in avtFilter::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#36 0x00002aaab0736c70 in avtDataObject::Update(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#37 0x00002aaab07be3e0 in avtTerminatingSink::Execute(ref_ptr<avtContract>) () from /g/g17/whitlocb/Development/trunk/src/lib/libavtpipeline_par.so
#38 0x00002aaaaebeff2e in avtPlot::Execute(ref_ptr<avtDataObject>, ref_ptr<avtContract>, WindowAttributes const*) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libavtplotter_par.so
---Type <return> to continue, or q <return> to quit--- 
#39 0x00002aaaab5addd7 in DataNetwork::GetWriter(ref_ptr<avtDataObject>, ref_ptr<avtContract>, WindowAttributes*) ()
   from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#40 0x00002aaaab5ee281 in NetworkManager::GetOutput(bool, bool, float*) () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#41 0x00002aaaab5d4b83 in EngineRPCExecutor<ExecuteRPC>::Execute(ExecuteRPC*) () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#42 0x00002aaab5bca78c in Subject::Notify() () from /g/g17/whitlocb/Development/trunk/src/lib/libvisitcommon.so
#43 0x00002aaab5ade02d in AttributeSubject::Notify() () from /g/g17/whitlocb/Development/trunk/src/lib/libvisitcommon.so
#44 0x00002aaaab6077a6 in MPIXfer::Process() () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#45 0x00002aaaab5d04a4 in Engine::EventLoop() () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#46 0x00002aaaab5d0a18 in Engine::PAR_EventLoop() () from /g/g17/whitlocb/Development/trunk/src/lib/libengine_par.so
#47 0x0000000000401a35 in EngineMain(int, char**) ()
#48 0x00002aaab8ed6d5d in __libc_start_main (main=0x401eb0 <main>, argc=9, ubp_av=0x7fffffffd0d8, init=<value optimized out>, 
    fini=<value optimized out>, rtld_fini=<value optimized out>, stack_end=0x7fffffffd0c8) at libc-start.c:226
#49 0x0000000000401819 in _start ()
</pre>

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Brad Whitlock
Original creation: 09/25/2015 02:41 am
Original update: 10/08/2015 01:35 pm
Ticket number: 2393
